helix-term/src/compositor.rs

```diff
@@ -21,6 +21,7 @@ pub type Callback = Box<dyn FnOnce(&mut Compositor, &mut Context)>;
 pub enum EventResult {
     Ignored,
     Consumed(Option<Callback>),
+    Used(Callback),
```

Comment:
Rather than another variant, we should turn `Ignored` into `Ignored(Option<Callback>)`.

Or the type could be `EventResult { ty: Ignored, callback: ... }`

Reply:
do we ever expect more variants?

What would you prefer:
```rust
pub struct EventResult {
    pub ty: EventResultType,
    pub callback: Option<Callback>
}

pub enum EventResultType {
    Ignored,
    Consumed
}
```

or

```
pub struct EventResult {
    pub is_consumed: bool,
    pub callback: Option<Callback>
}

```

or 

```rust
pub enum EventResultType {
    Ignored(Option<Callback>),
    Consumed(Option<Callback>)
}
```

Reply:
Let's do

```rust
pub enum EventResult {
    Ignored(Option<Callback>),
    Consumed(Option<Callback>)
}
```

because it's easier to construct